config/dictionaries/resource.json

```diff
@@ -288,22 +313,43 @@
       {
         "id": "move-device",
         "summary": "Move device to another application",
+        "description": "",
         "method": "PATCH",
         "endpoint": "/v5/device(<ID>)",
         "filters": "",
         "data": "{\n    \"belongs_to__application\": \"<APP ID>\"\n}"
+	  },
+	  {
+        "id": "get-target-release",
+        "summary": "Get the release device is pinned to",
+        "description": "Note: `should_be_running__release` will be unset or null if the device isn't pinned to any release. To find target release for an unpinned device, check the `is_on__commit` field example below instead.",
+        "method": "GET",
+        "endpoint": "/v5/device(<ID>)",
+        "filters": "?\\$select=should_be_running__release",
+        "data": ""
       },
       {
         "id": "set-device-to-release",
-        "summary": "Set device to a specific release",
+        "summary": "Pin device to a specific release",
+        "description": "",
         "method": "PATCH",
         "endpoint": "/v5/device(<ID>)",
         "filters": "",
         "data": "{\n    \"should_be_running__release\": \"<RELEASE ID>\"\n}"
+	  },
+	  {
+        "id": "get-current-release",
+        "summary": "Get current release running on device",
+        "description": "The `is_on__commit` field fetches the hash of the current release that the device is running on",
+        "method": "GET",
+        "endpoint": "/v5/device(<ID>)",
```

Comment:
@pdcastro Making a new thread,
Is there a reason why we have Release UUID, Release ID (Similarly DeviceID, Device UUID, application ID)? I didn't know we even had device ID tbh till I made the API call yesterday.

Reply:
I understand that the ID would change if we changed the database "engine", for example by upgrading to a new version of Postgres that had an incompatible database format, or if we changed from Postgres to MySQL, and the like. As such, the ID is an implementation detail that, ideally, would be hidden from users. By contrast, the UUID is eternally immutable, and would survive the database changes mentioned above.

See also this CLI issue: https://github.com/balena-io/balena-cli/issues/1843

Having said that, we have always exposed the numeric database IDs to end users via the balena API, we even document it in examples and projects and (to some extent) in the CLI, so at the moment the IDs are effectively part of the "public API". I suppose we could at any time have an API breaking change that hid database IDs, exposing UUIDs only, and then have a API translation layer that mapped "old" DB IDs to UUIDs, with a performance hit. This is my understanding...

@Page-, does this sound right?

Reply:
Coincidentally, a user has just created a new CLI issue where they describe using app IDs as a "security issue" that "an attacker could easily leverage" (their words!):

> https://github.com/balena-io/balena-cli/issues/2063
> As an aside, I'd even prefer that the app ids be internal only and not exposed to any customers or customer APIs. By itself this isn't a serious security issue, but it needlessly exposes info that an attacker could easily leverage.

Reply:
Also timely: https://github.com/balena-io/open-balena-api/pull/491/files

Reply:
@pdcastro are we lucky? or was this just bound to happen all at once :joy:

Reply:
Play the lottery! I suggest using the numbers of the issues and pull requests. 😄